File: backend/src/main/java/com/mhbilliards/billiards_management/service/shift/ShiftService.java
package com.mhbilliards.billiards_management.service.shift;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.mhbilliards.billiards_management.dto.shift.ShiftRequest;
import com.mhbilliards.billiards_management.dto.shift.ShiftResponse;

public interface ShiftService {
    ShiftResponse createShift(ShiftRequest request);

    ShiftResponse updateShift(Long id, ShiftRequest request);

    ShiftResponse getShiftById(Long id);

    Page<ShiftResponse> getAllShifts(Pageable pageable);

    void deleteShift(Long id);
}

File: frontend/src/services/shiftService.ts
import axiosClient from "./axiosClient";
import { ApiResponse, PageResponse, PaginationParams } from "@/types/base";
import { IShiftRequest, IShiftResponse } from "@/types/shift";

const API_URL = "/shifts";

const shiftService = {
    createShift: async (req: IShiftRequest): Promise<ApiResponse<null>> => {
        const res = await axiosClient.post(API_URL, req);
        return res.data;
    },

    updateShift: async (
        id: number,
        req: IShiftRequest,
    ): Promise<ApiResponse<null>> => {
        const res = await axiosClient.put(`${API_URL}/${id}`, req);
        return res.data;
    },

    getShiftById: async (id: number): Promise<ApiResponse<IShiftResponse>> => {
        const res = await axiosClient.get(`${API_URL}/${id}`);
        return res.data;
    },

    getAllShifts: async (
        params?: PaginationParams,
    ): Promise<ApiResponse<PageResponse<IShiftResponse>>> => {
        const res = await axiosClient.get(API_URL, {
            params: {
                page: params?.page ?? 0,
                size: params?.size ?? 20,
                sort: `${params?.sortBy ?? "id"},${params?.sortDirection ?? "asc"}`,
            },
        });
        return res.data;
    },

    deleteShiftById: async (id: number): Promise<ApiResponse<null>> => {
        const res = await axiosClient.delete(`${API_URL}/${id}`);
        return res.data;
    },
};

export default shiftService;
